onmt/utils/loss.py

```diff
@@ -35,14 +36,21 @@ def build_loss_compute(model, tgt_field, opt, train=True):
             len(tgt_field.vocab), opt.copy_attn_force,
             unk_index=unk_idx, ignore_index=padding_idx
         )
-    elif opt.label_smoothing > 0 and train:
+    elif opt.label_smoothing > 0:
         criterion = LabelSmoothingLoss(
-            opt.label_smoothing, len(tgt_field.vocab), ignore_index=padding_idx
+            opt.label_smoothing, len(tgt_field.vocab),
+            ignore_index=padding_idx
+        )
+    elif opt.unlikelihood_coeff > 0:
```

Comment:
Shall we assert in the opts validation that `unlikelihood_coeff` isn't compatible with `label_smoothing`?

Reply:
I can make them mutually exclusive with the parser or I can make both of them compatible at the same time as unlikelihood_coeff can be added to any loss (but label smoothing and unlikelihood are contradictory)

Reply:
Yes, let's go for the mutually exclusive way.